### PR TITLE
Update OpusControl.cs

### PR DESCRIPTION
**Pull request by Aaron2550**
Add official Documentation to each enum entry

### CODE REVIEW

File: src/Opus/OpusControl.cs [hunk 1]
```diff
@@ -1,48 +1,232 @@
...
+        /// </summary>
         SetApplication = 4000,
+        
+        /// <summary>
+        /// Gets the encoder's configured application.
+        /// </summary>
         GetApplication = 4001,
+        
+        /// <summary>
+        /// Configures the bitrate in the encoder. Rates from 500 to 512000 bits per second are meaningful, as well as the special values OPUS_AUTO and OPUS_BITRATE_MAX. The value OPUS_BITRATE_MAX can be used to cause the codec to use as much rate as it can, which is useful for controlling the rate by adjusting the output buffer size.
```
Comment (OoLunar) [inline src/Opus/OpusControl.cs]: Please `<see cref="OpusApplication.Auto"/>`, I'll figure out `OPUS_BITRATE_MAX` later.

File: src/Opus/OpusControl.cs [hunk 2]
```diff
@@ -1,48 +1,232 @@
...
+        /// </summary>
         SetBitrate = 4002,
+        
+        /// <summary>
+        /// Gets the encoder's bitrate configuration.
+        /// </summary>
         GetBitrate = 4003,
+        
+        /// <summary>
+        /// Configures the maximum bandpass that the encoder will select automatically. Applications should normally use this instead of OPUS_SET_BANDWIDTH (leaving that set to the default, OPUS_AUTO). This allows the application to set an upper bound based on the type of input it is providing, but still gives the encoder the freedom to reduce the bandpass when the bitrate becomes too low, for better overall quality.
```
Comment (OoLunar) [inline src/Opus/OpusControl.cs]: `cref` on `OPUS_SET_BANDWIDTH` and `OPUS_AUTO`

File: src/Opus/OpusControl.cs [hunk 3]
```diff
@@ -1,48 +1,232 @@
...
+        /// </summary>
         SetVbr = 4006,
+        
+        /// <summary>
+        /// Determine if variable bitrate (VBR) is enabled in the encoder.
+        /// </summary>
         GetVbr = 4007,
+        
+        /// <summary>
+        /// Sets the encoder's bandpass to a specific value. This prevents the encoder from automatically selecting the bandpass based on the available bitrate. If an application knows the bandpass of the input audio it is providing, it should normally use OPUS_SET_MAX_BANDWIDTH instead, which still gives the encoder the freedom to reduce the bandpass when the bitrate becomes too low, for better overall quality.
```
Comment (OoLunar) [inline src/Opus/OpusControl.cs]: `cref`

File: src/Opus/OpusControl.cs [hunk 4]
```diff
@@ -1,48 +1,232 @@
...
+        /// Gets the total samples of delay added by the entire codec. This can be queried by the encoder and then the provided number of samples can be skipped on from the start of the decoder's output to provide time aligned input and output. From the perspective of a decoding application the real data begins this many samples late.
+        /// The decoder contribution to this delay is identical for all decoders, but the encoder portion of the delay may vary from implementation to implementation, version to version, or even depend on the encoder's initial configuration. Applications needing delay compensation should call this CTL rather than hard-coding a value.
+        /// </summary>
         GetLookahead = 4027,
-        //ResetState = 4028, //Commented out in the reference Implementtation
+        
+        //ResetState = 4028, //Commented out in the reference Implementation.
+        
+        /// <summary>
+        /// Gets the sampling rate the encoder or decoder was initialized with. This simply returns the Fs value passed to opus_encoder_init() or opus_decoder_init().
```
Comment (OoLunar) [inline src/Opus/OpusControl.cs]: `cref`

File: src/Opus/OpusControl.cs [hunk 5]
```diff
@@ -1,48 +1,232 @@
...
+        /// <summary>
+        /// Gets the pitch of the last decoded frame, if available. This can be used for any post-processing algorithm requiring the use of pitch, e.g. time stretching/shortening. If the last frame was not voiced, or if the pitch was not coded in the frame, then zero is returned.
+        /// </summary>
+        /// <remarks>
+        /// This CTL is only implemented for decoder instances.
+        /// </remarks>
         GetPitch = 4033,
+        
+        /// <summary>
+        /// Configures decoder gain adjustment. Scales the decoded output by a factor specified in Q8 dB units. This has a maximum range of -32768 to 32767 inclusive, and returns OPUS_BAD_ARG otherwise. The default is zero indicating no adjustment. This setting survives decoder reset.
```
Comment (OoLunar) [inline src/Opus/OpusControl.cs]: `cref`

File: src/Opus/OpusControl.cs [hunk 6]
```diff
@@ -1,48 +1,232 @@
...
+        /// </summary>
+        /// <remarks>
+        /// gain = pow(10, x/(20.0*256))
```
Comment (OoLunar) [inline src/Opus/OpusControl.cs]: `<c>code</c>`

File: src/Opus/OpusControl.cs [hunk 7]
```diff
@@ -1,48 +1,232 @@
...
+        
+        /// <summary>
+        /// Gets the decoder's configured gain adjustment.
+        /// </summary>
+        GetGain = 4045, //Typo, should have been 4035
+        
+        /// <summary>
+        /// Configures the depth of signal being encoded.
+        /// This is a hint which helps the encoder identify silence and near-silence. It represents the number of significant bits of linear intensity below which the signal contains ignorable quantization or other noise.
+        /// For example, OPUS_SET_LSB_DEPTH(14) would be an appropriate setting for G.711 u-law input. OPUS_SET_LSB_DEPTH(16) would be appropriate for 16-bit linear pcm input with opus_encode_float().
```
Comment (OoLunar) [inline src/Opus/OpusControl.cs]: `cref` `OPUS_SET_LSB_DEPTH` and `opus_encode_float`

File: src/Opus/OpusControl.cs [hunk 8]
```diff
@@ -1,48 +1,232 @@
...
+        /// When using opus_encode() instead of opus_encode_float(), or when libopus is compiled for fixed-point, the encoder uses the minimum of the value set here and the value 16.
```
Comment (OoLunar) [inline src/Opus/OpusControl.cs]: `cref` `opus_encode` and `opus_encode_float`

File: src/Opus/OpusControl.cs [hunk 9]
```diff
@@ -1,48 +1,232 @@
...
+        /// </summary>
         GetLsbDepth = 4037,
+        
+        /// <summary>
+        /// Gets the duration (in samples) of the last packet successfully decoded or concealed.
+        /// </summary>
         GetLastPacketDuration = 4039,
+        
+        /// <summary>
+        /// Configures the encoder's use of variable duration frames. When variable duration is enabled, the encoder is free to use a shorter frame size than the one requested in the opus_encode*() call. It is then the user's responsibility to verify how much audio was encoded by checking the ToC byte of the encoded packet. The part of the audio that was not encoded needs to be resent to the encoder for the next call. Do not use this option unless you really know what you are doing.
```
Comment (OoLunar) [inline src/Opus/OpusControl.cs]: `cref` `opus_encode`, remove extra `*` too.

File: src/Opus/OpusControl.cs [hunk 10]
```diff
@@ -1,48 +1,232 @@
...
+        /// </summary>
         SetExpertFrameDuration = 4040,
+        
+        /// <summary>
+        /// Gets the encoder's configured use of variable duration frames.
+        /// </summary>
         GetExpertFrameDuration = 4041,
+        
+        /// <summary>
+        /// If set to 1, disables almost all use of prediction, making frames almost completely independent. This reduces quality.
```
Comment (OoLunar) [inline src/Opus/OpusControl.cs]: If set to `<see langword="true"/>`

File: src/Opus/OpusControl.cs [hunk 11]
```diff
@@ -1,48 +1,232 @@
...
+        
+        /// <summary>
+        /// Gets the encoder's configured prediction status.
+        /// </summary>
         GetPredictionDisabled = 4043,
+        
         //4045 is GetGain
+
+        /// <summary>
+        /// If set to 1, disables the use of phase inversion for intensity stereo, improving the quality of mono downmixes, but slightly reducing normal stereo quality. Disabling phase inversion in the decoder does not comply with RFC 6716, although it does not cause any interoperability issue and is expected to become part of the Opus standard once RFC 6716 is updated by draft-ietf-codec-opus-update.
```
Comment (OoLunar) [inline src/Opus/OpusControl.cs]: If set to `<see langword="true"/>`

**Review by OoLunar — APPROVED** [review]
Generally LGTM, just nitpicks. Up to you on if you want to fix them or not. Thanks for contributing.

File: src/Opus/OpusControl.cs (hunk 4 above)
Comment (Aaron2550) [reply src/Opus/OpusControl.cs]: To what? OpusEncoder.Create or OpusEncoder.Init? or maybe OpusNativeMethods.EncoderInit?

Comment (OoLunar) [reply src/Opus/OpusControl.cs]: It's exposed to the user, so `OpusEncoder.Init`

**Pull request merged**